### PR TITLE
DOC: update release document

### LINKED ISSUE #1238: updates to release.md
- [x] base this on `conda` environments instead of `virtualenv`
- [x] note that version strings may need to be changed in new `@experimental/@deprecated/@stable` code under _Prep the release_

### PR DESCRIPTION
**Pull request by jairideout**
Updated the release document to use conda instead of virtualenv. Added instructions for updating API stability decorator versions and removing deprecated functionality. Made other general improvements to the doc, including better CLI commands and clearer ordering of steps in the release process.

These new instructions roughly reflect the steps taken to build scikit-bio 0.5.1.

Fixes #1238.

Note: the diff is hard to follow. I recommend reading through the rendered doc.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -4,90 +4,81 @@
...
+4. Update ``CHANGELOG.md`` to include descriptions of all **user-facing** changes that made it into this release. Be sure to update the heading to include the new version (1.2.4) and the date of the release. Use the existing structure in the file as a template/guide.
 
-## Build the documentation
+5. Submit a pull request and merge when Travis-CI tests are passing.
 
-In the meantime, you can build the documentation and update the website.
+## Build website docs
 
-**Note:** You will need to **fully install** (including built extensions) the exact version of scikit-bio that you are editing so that Sphinx will pull docstrings from the correct version of the code. **Make sure the version of scikit-bio that is imported by ``import skbio`` is the correct one!**
+You will need to **fully install** the latest master branch of scikit-bio (including built extensions) and build the docs from this version. Make sure the version of scikit-bio that is imported by ``import skbio`` is the correct one before building the docs.
```
Comment (ebolyen) [inline RELEASE.md]: I would bold the last sentence, if only because I have messed this part up countless times myself.

File: RELEASE.md [hunk 2]
```diff
@@ -4,90 +4,81 @@
...
-
-4. Create a directory for the new version of the docs and recreate the ``latest/`` directory:
+3. Remove ``docs/latest``:
 
-        mkdir docs/1.2.4
-        mkdir docs/latest
+        git rm -rf docs/latest
 
-5. Copy over the built documentation to both ``docs/1.2.4/`` and ``docs/latest``:
+4. Copy over the built documentation to ``docs/1.2.4`` and ``docs/latest``:
```
Comment (ebolyen) [inline RELEASE.md]: Maybe change `1.2.4` to `x.y.z` so it's more clear that it's a placeholder (and it'll look weird if it shows up on the site)

File: RELEASE.md [hunk 3]
```diff
@@ -4,90 +4,81 @@
...
 
-        cp -r <path to skbio repo>/doc/build/html/* docs/1.2.4/
-        cp -r <path to skbio repo>/doc/build/html/* docs/latest/
+        cp -r doc/build/html docs/latest
+        cp -r doc/build/html docs/1.2.4
 
-6. Add a new list item to ``index.html`` to link to ``docs/1.2.4/index.html``.
+5. Add a new list item to ``index.html`` to link to ``docs/1.2.4/index.html``.
```
Comment (ebolyen) [inline RELEASE.md]: There's another step which is to copy most of the freshly rendered `README.md` into the index. In practice I did this via copying GitHub's source, but it really doesn't matter. We just probably want to leave the header alone (copy only the content).

File: RELEASE.md (hunk 1 above)
Comment (jairideout) [inline RELEASE.md]: Good idea, will do!

File: RELEASE.md (hunk 2 above)
Comment (jairideout) [inline RELEASE.md]: Good idea, will do!

File: RELEASE.md (hunk 3 above)
Comment (jairideout) [inline RELEASE.md]: I'll add a bullet noting that content from the readme should be ported to index.html. I remember that copying github's source stopped working at some point, so I won't specify exactly how to port the content.

**Comment by jairideout** [comment]
Thanks @ebolyen! I addressed your comments in a squash commit.
